sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RedirectPolicy.java

```diff
@@ -40,15 +40,19 @@ public RedirectPolicy(RedirectStrategy redirectStrategy) {
     @Override
     public HttpResponse process(HttpRequest httpRequest, HttpPipelineNextPolicy next) {
         // Reset the attemptedRedirectUrls for each individual request.
-        return attemptRedirect(httpRequest, next, 1, new HashSet<>());
+        try {
+            return attemptRedirect(httpRequest, next, 1, new HashSet<>());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
     }
 
     /**
      * Function to process through the HTTP Response received in the pipeline and redirect sending the request with a
      * new redirect URL.
      */
     private HttpResponse attemptRedirect(final HttpRequest httpRequest, final HttpPipelineNextPolicy next,
-                                         final int redirectAttempt, Set<String> attemptedRedirectUrls) {
+                                         final int redirectAttempt, Set<String> attemptedRedirectUrls) throws IOException {
```

Comment:
why are these throwing IOException now?

Reply:
It shouldn't. I think I missed it when merging `main` into this branch.

Reply:
I'll remove it.